### DISCUSSION TITLE
Privacy and security concernes

### DISCUSSION THREAD

**Discussion by giachat** [mayooear/ai-pdf-chatbot-langchain#142] [General]
Great project! I had a question in regards to privacy and security. Is uploaded pdf data in anyway sandboxed or protected? Have anyone thought how to implement this when it comes to proprietary company data?

Thanks!

**Comment by bschleter** [comment]
Correct me if I'm wrong anybody, but any data is what you may publish on your git repository to train, or what is used in questions/prompts to OpenAI/other LLM. OpenAI's data policy for API is private in sense it is not used to train the model, and kept only for [30 days it appears for criminal proceedings/law purposes](https://openai.com/policies/api-data-usage-policies). 

Theoretically, your text data is also stored into vector databases like pinecone, but it is stored in vectors with IDs, so not actually text until converted back and would need an LLM agent and some other info to convert back. This would only be the chain data to make the model, but this is what I assume is proprietary data initially. From what I know, text is converted to vectors before arriving to pinecone.

**Reply by gzajko** [reply]
it looks like there is actually full text in the pinecone database bundled as "metadata" with the vector. or am i wrong?

**Reply by 3koozy** [reply]
and by definition all embeddings should be universal m ie anyone with an embedding model like (openai) can transform them back to text.
